### DISCUSSION TITLE
Proxy problem

### DISCUSSION THREAD

**Discussion by fyrekrig** [stefansundin/privatkopiera#92] [Q&A]
I am outside Sweden and using proxy to watch SVT play and it works fine.
Now I want download programs if possible. I have installed your extension on Chrome and installed ffmpeg.
I am now in Chrome (through proxy ( 127.0.0.1: 61571 IP194.59.250.67)) at https://www.svtplay.se/video/25626436/varldens-natur-serengeti/varldens-natur-serengeti-avsnitt-1 and try to download. Copy command and get this message

C:\Windows\System32>ffmpeg -i "https://svt-vod-10g.akamaized.net/d0/se/20210824/d8e964f7-a5ad-4992-90ea-e3d5fc67d006/hls-video-avc-1920x1080p25-3089/hls-video-avc-1920x1080p25-3089.m3u8" -i "https://svt-vod-10g.akamaized.net/d0/se/20210824/d8e964f7-a5ad-4992-90ea-e3d5fc67d006/hls-audio-aac-2ch-192/hls-audio-aac-2ch-192.m3u8" -i "https://svt-vod-10g.akamaized.net/d0/se/20210824/d8e964f7-a5ad-4992-90ea-e3d5fc67d006/text/text-0.vtt" -vcodec copy -acodec copy "Världens natur- Serengeti - Avsnitt 1.mkv"
ffmpeg version 4.4-essentials_build-www.gyan.dev Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.2.0 (Rev6, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[https @ 00000233b7cbe840] HTTP error 403 Forbidden
https://svt-vod-10g.akamaized.net/d0/se/20210824/d8e964f7-a5ad-4992-90ea-e3d5fc67d006/hls-video-avc-1920x1080p25-3089/hls-video-avc-1920x1080p25-3089.m3u8: Server returned 403 Forbidden (access denied)

Your comment will be appreciated

**Comment by fyrekrig** [comment]
Sorry not an answer.
Managed to download a program that was allowed outside Sweden.
However can not find where the file is stored.

[https @ 000001ac8bc5d300] Opening 'https://svt-vod-10c.akamaized.net/d0/world/20210305/81e5bdce-d222-471e-93e3-720a53c54e81/hls-video-avc-4/hls-video-avc-4-1391.ts' for reading
[https @ 000001ac8cb77cc0] Opening 'https://svt-vod-10c.akamaized.net/d0/world/20210305/81e5bdce-d222-471e-93e3-720a53c54e81/hls-audio-stereo-1/hls-audio-stereo-1-1391.aac' for reading
[https @ 000001ac8bc2ab40] Opening 'https://svt-vod-10c.akamaized.net/d0/world/20210305/81e5bdce-d222-471e-93e3-720a53c54e81/hls-video-avc-4/hls-video-avc-4-1392.ts' for reading
[https @ 000001ac8c99c3c0] Opening 'https://svt-vod-10c.akamaized.net/d0/world/20210305/81e5bdce-d222-471e-93e3-720a53c54e81/hls-audio-stereo-1/hls-audio-stereo-1-1392.aac' for reading
[https @ 000001ac8bc5d300] Opening 'https://svt-vod-10c.akamaized.net/d0/world/20210305/81e5bdce-d222-471e-93e3-720a53c54e81/hls-video-avc-4/hls-video-avc-4-1393.ts' for reading
[https @ 000001ac8cb77cc0] Opening 'https://svt-vod-10c.akamaized.net/d0/world/20210305/81e5bdce-d222-471e-93e3-720a53c54e81/hls-audio-stereo-1/hls-audio-stereo-1-1393.aac' for reading
frame=133657 fps=149 q=-1.0 Lsize= 2147167kB time=01:29:06.30 bitrate=3290.0kbits/s speed=5.95x
video:2019125kB audio:127018kB subtitle:2kB other streams:0kB global headers:1kB muxing overhead: 0.047602%

So where will the downloaded file go?

**Comment by Bub0** [comment]
Try navigate to the folder where you want the files to be saved in, in the
command prompt, before you run the download command. Ex, "cd C:\Users\YOUR
USER \Downloads"
Remove " " and change YOUR USER to your user.
Otherwise you could run a search for the file name to locate the previously
downloaded file. :-)

Den fre 3 sep. 2021 08:41fyrekrig ***@***.***> skrev:

**Comment by stefansundin** [comment]
Hi @fyrekrig. As @Bub0 already described (thanks!), the files will be written to the directory that the cmd program "is currently in". It will display at the beginning of the command prompt line.

The website links to [an example image](https://stefansundin.github.io/privatkopiera/img/cmd.png). In this example, the path where the files will be written to is `C:\Användare\Nisse\Skrivbord` (the desktop folder on the Swedish version of Windows).

As for your proxy problems, can you give more information? If it is a system-wide VPN then I see no reason why it wouldn't work. There is a separate page on the website for other kinds of proxying that may perhaps be useful: https://stefansundin.github.io/privatkopiera/proxy.html

**Comment by fyrekrig** [comment]
Thanks your reply.  Had a look at your instructions for use via proxy.  I am not a programer so have difficulties getting the hang of how to set it up.
I don't have a VPN service. When I access SVT play for videos not allowed outside Sweden I use either psiphon3 (Swedish server) or AdvOr (Tor exit node Sweden)  Then I set Chrome to use these proxies. SVT Play now works fine!
I can get the proxy info and the Swedish IP but don't know/understand if I can use these with your program. 
If so can you give me a sample string I can try out.

**Comment by fyrekrig** [comment]
Tried this with Psison3

set http_proxy=http://194.59.250.67:49878/ffmpeg -i

ffmpeg -i "https://www.svtplay.se/video/26643218/bast-i-test/bast-i-test-sasong-4-bonusavsnitt?id=e5wyxrg"
ffmpeg version 4.4-essentials_build-www.gyan.dev Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.2.0 (Rev6, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[tcp @ 000002379ccd4440] Connection to tcp://194.59.250.67:49878 failed: Error number -138 occurred
https://www.svtplay.se/video/26643218/bast-i-test/bast-i-test-sasong-4-bonusavsnitt?id=e5wyxrg: Unknown error

Any comment appreciated

**Reply by stefansundin** [reply]
This seems wrong:

```
set http_proxy=http://194.59.250.67:49878/ffmpeg -i
```

I think you accidentally left `ffmpeg -i` in there?

So try:

```
set http_proxy=http://194.59.250.67:49878/
```

Searching the internet seems to indicate that "Error number -138 occurred" means that the TCP connection could not be established. So if this is the case then perhaps the proxy server is not up, or you have a firewall blocking that port?

**Comment by fyrekrig** [comment]
Tried again no luck.

I am on a Swedish IP and can open and play all restricted videos on SVT play.
When I activate extension "Privatekopiera" I am met with message "Fel.Den här hemsidan stöds ej"
Why? I am on a Swedish server!

I start playing the video for a short period and copy the URL and paste it to the Privatkopiera command.
And this is what happens.

set http_proxy=http://171.25.193.78:9050

ffmpeg -i "https://www.svtplay.se/video/32497207/draknastet/draknastet-sasong-1-avsnitt-5?id=85RwQXL"

ffmpeg -i "https://www.svtplay.se/video/32497207/draknastet/draknastet-sasong-1-avsnitt-5?id=85RwQXL"
ffmpeg version 4.4-essentials_build-www.gyan.dev Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.2.0 (Rev6, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[tcp @ 000001e05c764340] Connection to tcp://171.25.193.78:9050 failed: Error number -138 occurred
https://www.svtplay.se/video/32497207/draknastet/draknastet-sasong-1-avsnitt-5?id=85RwQXL: Unknown error

Same error as before. No firewall running. Server must be up as I can play the video.

**Comment by Bub0** [comment]
Paste this line into Ffmpeg. 🙂
![iMarkup_20210919_111254](https://user-images.githubusercontent.com/40185468/133922087-61f24661-c5e5-49a7-85a6-c11f938236a9.png)

**Comment by fyrekrig** [comment]
Kommando same as copy URL
![008](https://user-images.githubusercontent.com/29265156/133948910-31f69298-a60a-49c9-8295-b72aed61da5e.jpg)

**Reply by Bub0** [reply]
> Kommando same as copy URL
> ![008](https://user-images.githubusercontent.com/29265156/133948910-31f69298-a60a-49c9-8295-b72aed61da5e.jpg)
Yes it's the same, the button just copy it to the clipboard. 
This url you're in, in the picture won't work. Navigate to the episode you want instead! 
Ex, https://www.svtplay.se/video/32181751/draknastet/draknastet-sasong-1-avsnitt-1?info=visa
Best regards 👍

**Comment by fyrekrig** [comment]
Still not working.
Tried a few other proxies. Still no luck

set http_proxy=http://194.59.250.67:22
ffmpeg -i "https://svt-vod-4d.akamaized.net/d0/se/20210814/a2775c8d-606b-4c0d-b1bd-f605f7d06388/hls-video-avc-1920x1080p25-3099/hls-video-avc-1920x1080p25-3099.m3u8" -i "https://svt-vod-4d.akamaized.net/d0/se/20210814/a2775c8d-606b-4c0d-b1bd-f605f7d06388/hls-audio-aac-2ch-192/hls-audio-aac-2ch-192.m3u8" -i "https://svt-vod-4d.akamaized.net/d0/se/20210814/a2775c8d-606b-4c0d-b1bd-f605f7d06388/text/text-0.vtt" -vcodec copy -acodec copy "Draknästet - Avsnitt 1.mkv"
ffmpeg version 4.4-essentials_build-www.gyan.dev Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.2.0 (Rev6, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
https://svt-vod-4d.akamaized.net/d0/se/20210814/a2775c8d-606b-4c0d-b1bd-f605f7d06388/hls-video-avc-1920x1080p25-3099/hls-video-avc-1920x1080p25-3099.m3u8: End of file
